src/deployment/deploy.ts

```diff
@@ -20,40 +33,64 @@ async function setup() {
     provider: infuraProvider,
   };
 }
+
+async function deployContract(
+  deployerContract: Contract,
+  contractFactory: ContractFactory,
+  salt: string
+) {
+  const deployTx = contractFactory.getDeployTransaction();
+  const gas =
+    (await contractFactory.signer.provider!.estimateGas(deployTx)).toNumber() *
+    1.2;
+  const deployResponse = await deployerContract.deploy(deployTx.data, salt, {
+    gasLimit: gas,
+  });
+  await deployResponse.wait();
+  return getCreate2Address({
+    from: deployerAddress,
+    salt,
+    initCode: deployTx.data,
+  });
+}
+
 (async () => {
   // Set up wallets & provider
   const { admin } = await setup();
   console.log("Admin wallet address: " + admin.address);
+  const deployerContract = new Contract(deployerAddress, deployerABI, admin);
 
   const relayHubFactory = new RelayHubFactory(admin);
-  const tx = relayHubFactory.getDeployTransaction();
-  tx.gasPrice = parseEther("0.000000012");
-
-  const request = await admin.sendTransaction(tx);
-
-  const receipt = await request.wait(1);
-
-  console.log("Relay hub: " + receipt.contractAddress);
-
-  const proxyHubFactory = new ProxyAccountDeployerFactory(admin);
-  const proxyTx = proxyHubFactory.getDeployTransaction();
-
-  proxyTx.gasPrice = parseEther("0.000000012");
-  const proxyRequest = await admin.sendTransaction(proxyTx);
-  const proxyReceipt = await proxyRequest.wait(1);
+  const relayHubSalt = keccak256(toUtf8Bytes(VERSION + RELAY_HUB));
```

Comment:
why to the Utf8Bytes? Why not just do a normal encodePacked?

Reply:
normal encodePacked is actually more encoding that toUtf8Bytes. Internally encodePacked will call toUtf8Bytes to get the bytes representation of the string, then it will abi encode it. Is there a reason why we need to abi encode?

Reply:
ah ok. i was mostly thinking around standard approach. e.g. its typical to do an encodePacked before hashing in Ethereum. If toUtf8Bytes is basically doing that anyway then ok